### PR TITLE
In Pango, make dependencies to libthai, cairo, xft, fontconfig and freetype optional via meson options

### PR DESCRIPTION
**Pull request by Erlkoenig90**
Specify library name and version:  **pango/1.48.0**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.
 
To resurrect the [pull request](https://github.com/bincrafters/conan-pango/pull/3) to the old bincrafters repository, I'm submitting this to allow disabling various optional dependencies for pango if you don't need them. Since the corresponding changes to the meson system are already upstream, no patching in the recipe is necessary anymore. The package info now largely depends on the options instead of the OS for more flexibility.

### CODE REVIEW

File: recipes/pango/all/conanfile.py
Comment (prince-chrismc) [inline recipes/pango/all/conanfile.py:17]: please follow the convention of `with_dep` for the options

File: recipes/pango/all/conanfile.py
Comment (prince-chrismc) [inline recipes/pango/all/conanfile.py:44]: I love you C++

to sprinkle a little 🐍 

```suggestion
        if self.options.xft == "auto":
            self.options.xft = self.settings.os in ["Linux", "FreeBSD"]
        if self.options.fontconfig == "auto":
            self.options.fontconfig = not self.settings.os in ["Windows", "Macos"]
```

> ⚠️ typed manually might be a typo

File: recipes/pango/all/conanfile.py
Comment (prince-chrismc) [inline recipes/pango/all/conanfile.py:47]:
```suggestion
        if self.options.xft and not self.settings.os in ["Linux", "FreeBSD"]:
            raise ConanInvalidConfiguration("Xft can only be used on Linux and FreeBSD")
```

File: recipes/pango/all/conanfile.py
Comment (prince-chrismc) [inline recipes/pango/all/conanfile.py:59]: Should this option be deleted on Windows/Mac?

**Review by prince-chrismc — CHANGES_REQUESTED** [review]
If you could be so kind to apply https://github.com/conan-io/conan-center-index/blob/ddd8c3c7499678583061a570b03cac615d67b157/docs/reviewing.md#quotes

Please fix the option names

File: recipes/pango/all/conanfile.py
Comment (Erlkoenig90) [reply recipes/pango/all/conanfile.py]: Sorry, once a C++ programmer, always a C++ programmer :wink:

File: recipes/pango/all/conanfile.py
Comment (Erlkoenig90) [reply recipes/pango/all/conanfile.py:59]: Possibly, currently we throw a `ConanInvalidConfiguration` if Xft is set to `True` on Windows or Mac. Not sure what's more sensible.

Comment (prince-chrismc) [reply recipes/pango/all/conanfile.py:59]: Let's see if others comment

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/pango/all/conanfile.py
Comment (Erlkoenig90) [reply recipes/pango/all/conanfile.py]: Done in 541fe4e872319dbbb0b204b3f81cda03651d656c

**Comment by ericLemanissier** [comment]
macos fails with 
```
Run-time dependency cairo-win32 found: NO (tried pkgconfig, framework and cmake)
Run-time dependency cairo-quartz found: NO (tried pkgconfig, framework and cmake)

source_subfolder/meson.build:461:4: ERROR: Problem encountered: No Cairo font backends found
```
The cairo package seems to miss a component. I'm not sure why it passed before but not any more

**Comment by Erlkoenig90** [comment]
It looks like no `cairo-quartz` .pc files are being generated (`cairo-win32` is obviously unneeded):
```
pango/1.48.0: Generator pkg_config created cairo.pc
pango/1.48.0: Generator pkg_config created cairo-pdf.pc
pango/1.48.0: Generator pkg_config created cairo-ft.pc
pango/1.48.0: Generator pkg_config created cairo-fc.pc
pango/1.48.0: Generator pkg_config created cairo-gobject.pc
```
The [cairo recipe](https://github.com/conan-io/conan-center-index/blob/master/recipes/cairo/all/conanfile.py) doesn't even offer any `quartz` component or enabling building with `quartz`. Are we even sure that previously, pango was built with cairo support on Mac OS?

If not: We could disable cairo support by default, since it is now explicit (i.e. required or disabled) in the recipe instead of automatically determined by pango's meson script. When quartz support is added to the cairo recipe, that can then be reverted.

**Comment by ericLemanissier** [comment]
I'll add the cairo-quartz components as part of #4268, so this PR is blocked until it is merged.
(What changed on macos is that without your PR, pango is using fontconfig as a font backend, but your PR disables fontconfig by default on macos)

**Comment by Erlkoenig90** [comment]
Okay nice. Okay, I could easily re-enable fontconfig by default on Mac OS, if that's better

**Comment by ericLemanissier** [comment]
No I  think it's better to match upstream defaults. Let's do the actual fix

**Comment by prince-chrismc** [comment]
Please fix the failing hooks 🙏

**Comment by ericLemanissier** [comment]
@Erlkoenig90  can you retrigger CI now that #4268 is merged ? (edit PR description, or close it/reopen)
There is no hook error to correct here. `./bin/mesonbuild/cmake/data/preload.cmake` is part of meson, and is wrongly reported. The message will disappear when the build is successful.

**Comment by Erlkoenig90** [comment]
Alright, thanks for adding cairo-quartz!

File: recipes/pango/all/conanfile.py
Comment (ericLemanissier) [inline recipes/pango/all/conanfile.py:133]:
```suggestion
        if self.options.with_freetype and self.options.with_fontconfig:
```
see https://gitlab.gnome.org/GNOME/pango/-/blob/1.48.0/meson.build#L358

**Review by prince-chrismc — APPROVED** [review]

**Review by uilianries — COMMENTED** [review]
As "auto" means True or False, the package will be different, according the option value, but the package content will be same for "auto" and True (or False). The package ID should manipulated to avoid duplication. Take a look on this recipe as example: https://github.com/conan-io/conan-center-index/blob/885dbfdf0fb3096695511f6961339d85f505b7e2/recipes/getdns/all/conanfile.py#L134

**Comment by ericLemanissier** [comment]
configure makes sure that the value of the option is never "auto" : https://github.com/conan-io/conan-center-index/pull/4259/files#diff-cce23c7c070bb4c4744436deec65a16e1a0f7c106d3aeb5df57ca168e6865f1dR41-R46

**Review by uilianries — APPROVED** [review]

**Review by intelligide — APPROVED** [review]

**Pull request merged**
